strapi/strapi
# fix(content-releases): delete release-actions depending on changes to…

### What does it do?

Deletes release actions associated with a content-type if the content-type disables draft and publish or the contet-type is deleted.

Implementation details open to discussion:
- I used the eventHub (as opposed to db.lifecycles) to listen for the events above
- Since the function to delete release-actions for a given contentType is not specific to a release I decided to put this in a new release-action service.

### Why is it needed?

- Otherwise releases are broken when these changes happen

### How to test it?

- Create a release
- Add entries from different content-types
- Disable draft and publish on one of the content-types
- Go back to your release
- You should only see entries for the content-type that still has draft and publish
- Do the same test for deleting a content-type
- Do the same tests where entries exist in multiple releases

### Related issue(s)/PR(s)

CS-470
